src/Text/Parsing/StringParser.purs

```diff
@@ -26,26 +26,25 @@ type Pos = Int
 type PosString = { str :: String, pos :: Pos }
 
 -- | The type of parsing errors.
-newtype ParseError = ParseError String
-
-instance showParseError :: Show ParseError where
-  show (ParseError msg) = msg
-
-derive instance eqParseError :: Eq ParseError
-
-derive instance ordParseError :: Ord ParseError
+type ParseError = { error :: String, pos :: Pos }
 
 -- | A parser is represented as a function which takes a pair of
 -- | continuations for failure and success.
-newtype Parser a = Parser (PosString -> Either { pos :: Pos, error :: ParseError } { result :: a, suffix :: PosString })
+newtype Parser a = Parser (PosString -> Either ParseError { result :: a, suffix :: PosString })
 
 -- | Run a parser by providing success and failure continuations.
```

Comment:
These documentation comments seem a bit funky to me; they keep talking about continuations, but that's not what I interpret from this function. Am I missing something?

Reply:
I was thinking the same thing.

Reply:
Yeah, I think this can be removed.